backend/src/Machine/machine.controller.ts

```diff
@@ -0,0 +1,134 @@
+import { Router, Request, Response } from 'express';
+import MachineService from './machine.service';
+import passport from 'passport';
+import { requireParams, isNotNUllUndefinedEmpty } from '../shared/middleware/paramsChecker';
+
+class Controller {
+    public path = '/machine';
+    public router = Router();
+    private machineService: MachineService;
+
+    constructor(machineService: any = null) {
+        this.machineService = machineService || new MachineService();
+        this.initRoutes();
+    }
+
+    private initRoutes() {
+        // Retrieve all machines.
+        this.router.get(
+            '/',
+            passport.authenticate('jwt', { session: false }),
+            async (req: Request, res: Response) => {
+                const result = await this.machineService.getAllMachines();
+                res.json(result);
+            }
+        );
+
+        // Create new machine.
+        this.router.post(
+            '/',
+            passport.authenticate('jwt', { session: false }),
+            async (req: Request, res: Response) => {
+                const { status, numberOrderCompleted } = req.body;
+                var message;
+
+                try {
+                    const result = await this.machineService.createNewMachine();
+                    message = {
+                        id: result,
+                        status: true,
+                        message: 'Machine was created successfully.'
+                    };
+                } catch (e) {
+                    message = {
+                        status: false,
+                        message: e as string
+                    };
+                }
+
+                res.json(message);
+            }
+        );
+
+        // Retrieve a single machine with machineId.
+        this.router.get(
+            '/:machineId',
+            passport.authenticate('jwt', { session: false }),
+            async (req: Request, res: Response) => {
+                const id = Number(req.params.machineId);
+                const result = await this.machineService.findMachineById(id);
+                res.json(result);
+            }
+        );
+
+        // Update a machine with machineId.
+        this.router.put(
+            '/:machineId',
+            passport.authenticate('jwt', { session: false }),
+            async (req: Request, res: Response) => {
+                const id = Number(req.params.machineId);
+                const { status, numberOrderCompleted } = req.body;
```

Comment:
Why do you let the client decide on the number of orders completed instead of adding 1 to the current number of orders completed?

Why don't you abstract the concept of updating a machine by creating routes to either start or complete an order? That way the client only has to call either start or complete with a machine id.